packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -333,6 +336,66 @@ const ToggleMenuDrawer = observer(({ platform_config }) => {
         );
     };
 
+    const getLanguageRoutesTraderHub = () => {
+        const LanguageLink = ({ lang }) => (
+            <React.Fragment>
+                <Icon
+                    icon={`IcFlag${lang.replace('_', '-')}`}
+                    className={'settings-language__language-link-flag settings-language__language-flag--pre-appstore'}
+                    type={lang.replace(/(\s|_)/, '-').toLowerCase()}
+                />
+                <span
+                    className={classNames('settings-language__language-name', {
+                        'settings-language__language-name--active': currentLanguage === lang,
+                    })}
+                >
+                    {getAllowedLanguages()[lang]}
+                </span>
+            </React.Fragment>
+        );
+
+        return (
+            <MobileDrawer.SubMenu
+                is_expanded={is_language_changing}
+                has_subheader
+                submenu_title={localize('Language')}
+                onToggle={is_expanded => {
+                    expandSubMenu(is_expanded);
+                    setIsLanguageChanging(!is_language_changing);
+                }}
+                submenu_toggle_class='dc-mobile-drawer__submenu-toggle--hidden'
+            >
+                <div
+                    className={classNames('settings-language__language-container', {
+                        'settings-language__language-container--pre-appstore': is_pre_appstore,
+                    })}
+                >
+                    {Object.keys(getAllowedLanguages()).map(lang => (
+                        <span
+                            id={`dt_settings_${lang}_button`}
+                            key={lang}
+                            onClick={async () => {
+                                await changeLanguage(lang, changeCurrentLanguage);
+                                await i18n.changeLanguage(lang);
+                                toggleDrawer();
+                                setIsLanguageChanging(!is_language_changing);
+                            }}
```

Comment:
Better move it to its own function and invoke it here. too long to be an inline function.